File: CivitasJava/civitas/Sorpresa.java
package civitas;

import java.util.ArrayList;

public abstract class Sorpresa {
    
    // Atributos
    private String texto;
    
    // Métodos
    Sorpresa (String tex){
        texto = tex;
    }
    
    void informe (int actual, ArrayList<Jugador> todos){
        if ( jugadorCorrecto(actual, todos) ){
            String str = "Aplicando sorpresa " + this.toString() + " al jugador " + todos.get(actual).getNombre() + "\n";
            Diario.getInstance().ocurreEvento(str);
        }     
    }
    
    abstract void aplicarAJugador (int actual, ArrayList<Jugador> todos) throws Exception;
        
    public boolean jugadorCorrecto (int actual, ArrayList<Jugador> todos){
        return (actual < todos.size()) && (actual >= 0);
    }
    
    @Override
    public String toString (){
        return texto;
    };
}

File: CivitasRuby/sorpresa.rb
require_relative 'diario'

module Civitas
class Sorpresa
    
  # Constructor  
  def initialize(texto)
    @texto = texto
  end
    
  # Métodos    
  def informe(actual, todos)
    evento = "Se ha aplicado una sorpresa " + to_string + " al jugador " + todos[actual].get_nombre + "\n"
    Diario.instance.ocurre_evento(evento)
  end
    
  def jugador_correcto(actual, todos)
    return actual < todos.length
  end
    
  def to_string
    @texto
  end
  
  private_class_method :new
end
end